classifier_and_summarizer/summarizer/extractor_output_preprocessor.py

```diff
@@ -0,0 +1,224 @@
+
+
+class SentenceWithAttributes:
+    ''' Object to represent a summary sentence
+    or title sentence along with attributes such as
+    embedding or font_style
+    '''
+
+    def __init__(self, text, index, font_style, embedding):
+        self.text = text
+        self.index = index
+        self.font_style = font_style
+        self.embedding = embedding
+
+
+class ExtractorOutputPreprocessor:
+    ''' Class to implement the utilities for
+    preprocessing the extractor output
+
+    Preprocessing steps applied:
+    * Split contents into different types
+    * Assign img_description_embeddings to media
+    * Summarize the text content
+    '''
+
+    def __init__(self, contents):
+        self.content_list = contents.content_list
+        self.normal_text_content_list = list()  # non-title text
+        self.title_text_content_list = list()  # title text
+        self.media_content_list = list()  # images/gifs
+        self.embedded_content_list = list()  # insta/tweets/quotes
+        self.text_summarizer = TextSummarizer(priority="speed")
+        self.sentence_embedding_model \
+            = SentenceTransformer('bert-base-nli-stsb-mean-tokens')
+
+    def get_preprocessed_content_lists(self):
+
+        #  first split content into different categories
+        self._split_content()
+
+        # summarize the webpage text
+        self._summarize_text_content()
+
+        # assemble the sentence objects
+        self._make_sentence_objects()
+
+        # add image description embeddings
+        self._add_media_description_and_attribute_embeddings()
+
+        return {
+            "titles": self.title_text_content_list,
+            "sentences": self.sentence_objects_list,
+            "media": self.media_content_list,
+            "embedded_content": self.embedded_content_list
+        }
+
+    def _split_content(self):
+        ''' Splits the given content into 4 lists
+
+        * list of title text
+        * list of normal text - each text is a
+            paragraph or a block of sentences
+        * list of media content (images/gifs)
+        * list of embedded content
+        '''
+        for content in self.content_list:
+            if content.content_type == ContentType.TEXT:
+                if content.type == "title":
+                    self.title_text_content_list.append(content)
+                else:
+                    self.normal_text_content_list.append(content)
+
+            elif content.content_type == ContentType.IMAGE:
+                self.media_content_list.append(content)
+
+            elif content.content_type != ContentType.UNKNOWN:
+                self.embedded_content_list.append(content)
+
+    def _summarize_text_content(self):
+        '''
+        Applies text summarization to the combined text in the webpage
+        '''
+        # aggregate all text from the webpage
+        webpage_text = '. '.join(
+            [text.text_string for text in self.normal_text_content_list])
```

Comment:
Ensure that the summarizer be able to handle cases like "This is a ball.. This is a bat.". (that is, there might be cases where 2 '.' occur together)